docs/how-to-migrate-from-smartlock.md

```diff
@@ -0,0 +1,791 @@
+  tokenProviders.put(
+      "https://accounts.google.com",
+      new TokenRequestInfo.Builder()
+          .setClientId(MY_GOOGLE_CLIENT_ID)
+          .setNonce("randomlyGeneratedNonce")
+          .build());
+
+  Intent hintIntent = mCredentialClient.getHintRetrieveIntent(
+      new HintRetrieveRequest.Builder(
+          AuthenticationMethods.EMAIL,
+          AuthenticationMethods.PHONE,
+          AuthenticationMethods.GOOGLE)
+          .setTokenProviders(tokenProviders)
+          .build());
+
+  startActivityForResult(hintIntent, RC_HINT);
+}
+```
+
+The key differences to note are:
+
+- The `CredentialClient` is easier to create than a `GoogleApiClient` instance:
+  a `Context` reference is all that is required, with no asynchronous
+  initialization.
+
+- Account types from the Smart Lock for Passwords API are now “authentication
+  methods”, with email auth, phone auth and federated authentication methods
+  all treated in a consistent manner.
+
+- A plain `Intent` is used instead of a `PendingIntent` - we have found this to
+  be less confusing for developers, and the additional functionality provided by
+  PendingIntent proved to be unnecessary.
+
+One may also notice the new `setTokenProviders` method - in an effort to support
+acquiring ID tokens from additional providers, clients are now encouraged to
+explicitly declare the ID token providers they support (e.g. Google,
+Microsoft), along with any provider-specific parameters that may be required.
+Currently, OpenYOLO directly supports specifying your application’s OpenID
+Connect client ID and an ID token nonce value.
+
+### Receiving a hint response
+
+Hint responses are delivered to the application via onActivityResult in both
+the Smart Lock and OpenYOLO APIs. Both indicate the success or failure of the
+request via a result code and Intent extras, though the data is structurally
+different.
+
+With Smart Lock for Passwords:
+
+```java
+@Override protected void onActivityResult(
+    int requestCode, int resultCode, Intent data) {
+
+  if (requestCode == RC_HINT && resultCode == RESULT_OK) {
+    Credential credential =
+        data.getParcelableExtra(Credential.EXTRA_KEY);
+    if (credential == null) {
+      authenticateManually();
+    } else {
+      processSmartLockCredentialAsHint(credential);
+    }
+  }
+}
+```
+
+With OpenYOLO:
+
+```java
+@Override
+protected void onActivityResult(
+    int requestCode, int resultCode, Intent data) {
+
+  if (requestCode == RC_HINT) {
+    HintRetrieveResult result =
+        mCredentialClient.getHintRetrieveResult(data);
+
+    if (result.getResultCode() !=
+        HintRetrieveResult.CODE_HINT_SELECTED) {
+      authenticateManually();
+      return;
+    }
+
+    Hint hint = result.getHint();
+    processOpenYoloHint(hint);
+  }
+}
+```
+
+With OpenYOLO, the `resultCode` value will always be equal to the value returned
+by `result.getResultCode()`, so either can be used. The standard result codes
```

Comment:
IMO we shouldn't promote using the result code returned via onActivityResult(). If the provider's activity finishes without setting a result or crashes the result codes may not match.

Reply:
Good point, I'll remove this.